### PR TITLE
Update InitDatabase for rpc debugging

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

File: src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitDatabase.cs [hunk 1]
```diff
@@ -64,11 +68,11 @@ public async Task Execute()
                 _context.NodeDataConsumer = beamSyncProvider.NodeDataConsumer;
             }
 
-            // IDbProvider debugRecorder = new RocksDbProvider(Path.Combine(_context._initConfig.BaseDbPath, "debug"), dbConfig, _context._logManager, _context._initConfig.StoreTraces, _context._initConfig.StoreReceipts);
-            // _context._dbProvider = new RpcDbProvider(_context._jsonSerializer, new BasicJsonRpcClient(KnownRpcUris.Localhost, _context._jsonSerializer, _context._logManager), _context._logManager, debugRecorder);
-
-            // IDbProvider debugReader = new ReadOnlyDbProvider(new RocksDbProvider(Path.Combine(_context._initConfig.BaseDbPath, "debug"), dbConfig, _context._logManager, _context._initConfig.StoreTraces, _context._initConfig.StoreReceipts), false);
-            // _context._dbProvider = debugReader;
+            // RocksDbProvider debugRecorder = new RocksDbProvider(_context.LogManager);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitDatabase.cs]: is there a way to do that with some config switch so we do not have to comment out / comment  back in?

**Comment by tkstanczak** [comment]
Can we cover this as part of the delivery?
image

File: src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitDatabase.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitDatabase.cs]: Added diagnostic enumeration property

File: src/Nethermind/Nethermind.Runner/IInitConfig.cs [hunk 2]
```diff
@@ -62,7 +62,21 @@ public interface IInitConfig : IConfig
...
+        string RpcDbUrl { get; set; }
+    }
+    
+    public enum DiagnosticMode
+    {
+        None,
+        [ConfigItem(Description = "Diagnostics mode which uses an in-memory DB")]
+        MemDb,
+        [ConfigItem(Description = "Diagnostics mode which uses an remote DB")]
+        RpcDb,
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/IInitConfig.cs]: typo: a remote

File: src/Nethermind/Nethermind.Runner/IInitConfig.cs [hunk 3]
```diff
@@ -62,7 +62,21 @@ public interface IInitConfig : IConfig
...
+        [ConfigItem(Description = "Diagnostics mode which uses an read-only DB")]
+        ReadOnlyDb
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/IInitConfig.cs]: typo: a resd-only

**Comment by LukaszRozmej** [comment]
Only typo fixes

**Pull request merged**